[\#386 Issue](https://github.com/rear/rear/issues/386) `closed`: Issue in HP Data Protector restore script
==========================================================================================================

**Labels**: `bug`

#### <img src="https://avatars.githubusercontent.com/u/7138166?u=8badbe18661388be2a5be9c8dd8ca530bcdcbe50&v=4" width="50">[revog](https://github.com/revog) opened issue at [2014-04-02 12:04](https://github.com/rear/rear/issues/386):

Hi there

Currently I'm implementing ReaR in relation to HP Data Protector on
CentOS 6 machines.  
Generation of ISO worked fine. During reverse engineering of failed
disaster recovery process I got stuck on the
`restore/../40_restore_with_dp.sh` script.

I've seen, that we iterate through available sessions, backuped hosts
and file systems. After they get stored under /tmp/(dp\_)\*.

Within the script `verify/../50_select_dp_restore.sh` on line 30 we
query the available filesystems and store them in
`/tmp/dp_list_of_fs_objects`:

    ${OMNIDB} -filesystem | grep "${HOST}" | cut -d"'" -f -2 > /tmp/dp_list_of_fs_objects

But in a later step (40\_restore\_with\_dp.sh) we read in the file
`/tmp/list_of_fs_objects` which contains ALL available filesystems of
all backuped servers. This leads to the fact, that rear tries to restore
the first in the list (which does not belong to the client beeing
restored in my example) and fails.

Am I doing something wrong or shouldn't the `40_restore_with_dp.sh`
script take the `/tmp/dp_list_of_fs_objects` list instead of
`/tmp/list_of_fs_objects`?

Thanks for helping.

Regards

#### <img src="https://avatars.githubusercontent.com/u/888633?u=cdaeb31efcc0048d3619651aa18dd4b76e636b21&v=4" width="50">[gdha](https://github.com/gdha) commented at [2014-04-02 16:09](https://github.com/rear/rear/issues/386#issuecomment-39349475):

@revog it looks your re-engineering work was the nail on the head ;-)  
looking forward to your pull request. Thanks for the debugging.

#### <img src="https://avatars.githubusercontent.com/u/888633?u=cdaeb31efcc0048d3619651aa18dd4b76e636b21&v=4" width="50">[gdha](https://github.com/gdha) commented at [2014-04-03 10:40](https://github.com/rear/rear/issues/386#issuecomment-39436905):

@revog could you verify it works well and close the issue if possible?

#### <img src="https://avatars.githubusercontent.com/u/7138166?u=8badbe18661388be2a5be9c8dd8ca530bcdcbe50&v=4" width="50">[revog](https://github.com/revog) commented at [2014-04-03 13:42](https://github.com/rear/rear/issues/386#issuecomment-39452037):

Thx @gdha for your prompt support and merge. Just tested and it works
like a charm! cheers

------------------------------------------------------------------------

\[Export of Github issue for
[rear/rear](https://github.com/rear/rear).\]
